### PR TITLE
fix spurious test failures in resilience_move tests

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Fix spurious test failures in `resilience_move` tests (only happened in devel until now).

- [x] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

#### Related Information

- [x] There is a *JIRA Ticket number* (In case a customer was affected / involved): https://arangodb.atlassian.net/browse/BTS-42 

### Testing & Verification

This change is already covered by existing tests, such as *resilience_move*.

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9728/

### CODE REVIEW

**Comment by jsteemann** [comment]
Tests blue

File: arangod/RocksDBEngine/RocksDBReplicationManager.cpp [hunk 1]
```diff
@@ -305,6 +306,28 @@ void RocksDBReplicationManager::drop(TRI_vocbase_t* vocbase) {
   garbageCollect(false);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief drop contexts by collection (at least mark them as deleted)
+////////////////////////////////////////////////////////////////////////////////
+
+void RocksDBReplicationManager::drop(LogicalCollection* collection) {
+  LOG_TOPIC("fe4bb", TRACE, Logger::REPLICATION)
```
Comment (mchacki) [inline arangod/RocksDBEngine/RocksDBReplicationManager.cpp:314]: I know it is not called like this, but I would like to see an ASSERTION that collection is not a nullptr.
Just to be save in future

Comment (jsteemann) [reply arangod/RocksDBEngine/RocksDBReplicationManager.cpp:314]: As this PR is for 3.6, I will add the assertion to devel now.

Comment (jsteemann) [reply arangod/RocksDBEngine/RocksDBReplicationManager.cpp:314]: Added via 425d3fd0d9

Comment (mchacki) [reply arangod/RocksDBEngine/RocksDBReplicationManager.cpp:314]: Nice me much like :+1:

**Pull request merged**